helix-term/src/ui/editor.rs

```diff
@@ -228,7 +228,43 @@ impl EditorView {
             }
         }
 
-        // render selections
+        // render gutters
+
+        let linenr: Style = theme.get("ui.linenr");
+        let warning: Style = theme.get("warning");
+        let error: Style = theme.get("error");
+        let info: Style = theme.get("info");
+        let hint: Style = theme.get("hint");
+
+        for (i, line) in (view.first_line..last_line).enumerate() {
+            use helix_core::diagnostic::Severity;
+            if let Some(diagnostic) = doc.diagnostics().iter().find(|d| d.line == line) {
+                surface.set_stringn(
+                    viewport.x - OFFSET,
+                    viewport.y + i as u16,
+                    "●",
+                    1,
+                    match diagnostic.severity {
+                        Some(Severity::Error) => error,
+                        Some(Severity::Warning) | None => warning,
+                        Some(Severity::Info) => info,
+                        Some(Severity::Hint) => hint,
+                    },
+                );
+            }
+
+            // line numbers having selections are rendered differently
+            surface.set_stringn(
+                viewport.x + 1 - OFFSET,
+                viewport.y + i as u16,
+                format!("{:>5}", line + 1),
+                5,
+                linenr,
+            );
+        }
+
+        // render selections and selected linenr(s)
+        let linenr_select: Style = theme.get("ui.linenr.selected");
```

Comment:
This is a welcome change, but will break the color in themes that don't define the key (it currently has a strong colored fallback so you can notice a missing key): https://github.com/helix-editor/helix/blob/f33aaba53fee32fb0a6e496c472f4e08bb401237/helix-view/src/theme.rs#L206

I think a Option<>-able method needs to be added to `Theme` without the fallback, then you can `theme.try("ui.linenr.selected").unwrap_or(theme.get("ui.linenr"))`

Reply:
I'll add a `try` method to `Theme`. Though in the same way, shouldn't we handle proper fallbacks for all keys ? For example `ui.text` and `ui.text.focus` (and more in the future presumably).

Also a `try` method is handy for one-off shots but we would end up repeating the same `try` calls in multiple places as the UI gets more themable - in that case should we handle fallbacks in `Theme` itself ?